src/task/ReceiveVideoInputTask.ts

```diff
@@ -24,40 +23,35 @@ export default class ReceiveVideoInputTask extends BaseTask {
     height: number,
     frameRate: number
   ): Promise<void> {
-    const trackSettings = mediaStreamTrack.getSettings();
-    let videoQualitySettings: VideoQualitySettings;
+    const videoQualitySettings = isContentAttendee
+      ? this.context.meetingSessionConfiguration.meetingFeatures.contentMaxResolution
+      : this.context.meetingSessionConfiguration.meetingFeatures.videoMaxResolution;
 
-    if (isContentAttendee) {
-      videoQualitySettings = this.context.meetingSessionConfiguration.meetingFeatures
-        .contentMaxResolution;
-    } else {
-      videoQualitySettings = this.context.meetingSessionConfiguration.meetingFeatures
-        .videoMaxResolution;
-    }
     if (
-      width > videoQualitySettings.videoWidth ||
-      height > videoQualitySettings.videoHeight ||
-      frameRate > videoQualitySettings.videoFrameRate
+      !isContentAttendee &&
+      width <= videoQualitySettings.videoWidth &&
+      height <= videoQualitySettings.videoHeight &&
+      frameRate <= videoQualitySettings.videoFrameRate
     ) {
-      const constraint: MediaTrackConstraints = {
-        width: { ideal: videoQualitySettings.videoWidth },
-        height: { ideal: videoQualitySettings.videoHeight },
-        frameRate: { ideal: videoQualitySettings.videoFrameRate },
-      };
-      this.context.logger.warn(
-        `Video track (content = ${isContentAttendee}) will be constrained to: ${JSON.stringify(
-          constraint
-        )} to remain below configured video quality settings, trackSettings: ${JSON.stringify(
-          trackSettings
-        )}`
+      // Skip applying constraints if the current video track settings are already
```

Comment:
Any downside to just setting these on camera as well? Or maybe we can just consider this reducing impact of the change.

Reply:
Yeah, I would say the reason is to reduce impact. In fact, I noticed applying constraint to camera may make the browser insist to scale it to 640x480 when resolution is configured low. It never happened to content share (I tried a lot of different resolutions).